eng/pipelines/templates/jobs/smoke.tests.yml

```diff
@@ -159,7 +159,7 @@ jobs:
           timeoutInMinutes: 5
 
         - pwsh: |
-            $packages = Get-ChildItem "$(Pipeline.Workspace)/${{ parameters.ArtifactName }}/*.zip"
+            $packages = Get-ChildItem "$(Pipeline.Workspace)/${{ parameters.ArtifactName }}/*/*.zip"
```

Comment:
Can we also add add some assertions here to make sure we end up with the packages we think we should have? Or perhaps we switch from doing a listing to building the exact path which we should know now after we did the work to standardize how the artifacts are structured.

Reply:
Yeah, we do know exactly which version we want from the artifacts. Here's one when it can't find the correct package - https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1050076&view=logs&j=811de798-cac6-5a82-8dac-ad597ae5b1be&t=70ccc55a-509b-52ea-a60f-0b83826d1abf&l=166

Reply:
That is an indirect failure. We can make this step fail if the file package we know we are expecting doesn't exist.

Reply:
I'm not quite sure what you mean by this. There's a delay in the publishing of the newest versioned package and therefore we need the retries, in that case wouldn't that be sufficient to check for this?

Reply:
My main point here is that we know exactly what package(s) we are expecting to find from the build and if we don't find them then we should probably break here so that we can likely give a nicer error message.

Reply:
Actually taking a closer look I don't think we have really scoped this down to a single package yet as it looks to be trying to update all the packages we built. Did you plan do that as a separate change?